### ISSUE TITLE
feature/query_cxx_types: Type inconsistency found in BUFR variables with the same type

### ISSUE THREAD

**Issue by emilyhcliu** [JCSDA-internal/ioda-converters#977]
## Current behavior (describe the bug)
 
For ADPSFC PREPBUFR file ([bufr_ncep_prepbufr_adpsfc.yaml](https://github.com/JCSDA-internal/ioda-converters/blob/feature/query_cxx/test/testinput/bufr_ncep_prepbufr_adpsfc.yaml))

There are 7 quality markers: ZQM, PQM, PMQ, TQM, QQM, SSTQM, WQM, WQM  (for height, pressure, pressure reduced to mean sea level, temperature, moisture, SST, and winds, respectively)

The first fours (ZQM, PQM, PMQ, TQM) are converted to integer type.
However the rest of the markers (QQM, SSTQM, WQM) are converted to float. 

They should all be converted to integer.
The BUFR table indicates that the unit of these quality markers is CODE TABLE.  And all quality markers point to the same [code table](https://www.emc.ncep.noaa.gov/mmb/data_processing/prepbufr.doc/table_7.htm).

Query part:
```
          # QualityMarker
          heightQualityMarker:
            query: "*/Z___INFO/Z__EVENT/ZQM"
          pressureQualityMarker:
            query: "*/P___INFO/P__EVENT/PQM"
          pressureReducedToMeanSeaLevelQualityMarker:
            query: "*/PMSL_SEQ/PMQ"
          airTemperatureQualityMarker:
            query: "*/T___INFO/T__EVENT/TQM"
          specificHumidityQualityMarker:
            query: "*/Q___INFO/Q__EVENT/QQM"
          waterTemperatureQualityMarker:
            query: "*/SST_INFO/SSTEVENT/SSTQM"
          windEastwardQualityMarker:
            query: "*/W___INFO/W__EVENT/WQM"
          windNorthwardQualityMarker:
            query: "*/W___INFO/W__EVENT/WQM"

```
Export part:

```
group: QualityMarker {
  variables:
        uint airTemperature(Location, TemperatureEvent) ;
                airTemperature:_FillValue = 1215750144U ;
                string airTemperature:long_name = "Temperature Quality Marker" ;
                string airTemperature:coordinates = "longitude latitude" ;
        uint height(Location, HeightEvent) ;
                height:_FillValue = 1215750144U ;
                string height:long_name = "Height Quality Marker" ;
                string height:coordinates = "longitude latitude" ;
        uint pressure(Location, PressureEvent) ;
                pressure:_FillValue = 1215750144U ;
                string pressure:long_name = "Pressure Quality Marker" ;
                string pressure:coordinates = "longitude latitude" ;
        uint pressureReducedToMeanSeaLevel(Location) ;
                pressureReducedToMeanSeaLevel:_FillValue = 1215750144U ;
                string pressureReducedToMeanSeaLevel:long_name = "Mean Sea Level Pressure Quality Marker" ;
                string pressureReducedToMeanSeaLevel:coordinates = "longitude latitude" ;
        float specificHumidity(Location, HumidityEvent) ;
                specificHumidity:_FillValue = 1.e+11f ;
                string specificHumidity:long_name = "Specific Humidity Quality Marker" ;
                string specificHumidity:coordinates = "longitude latitude" ;
        float waterTemperature(Location, WaterTemperatureEvent) ;
                waterTemperature:_FillValue = 1.e+11f ;
                string waterTemperature:long_name = "Water Temperature Quality Marker" ;
                string waterTemperature:coordinates = "longitude latitude" ;
        float windEastward(Location, WindEvent) ;
                windEastward:_FillValue = 1.e+11f ;
                string windEastward:long_name = "U, V-Component of Wind Quality Marker" ;
                string windEastward:coordinates = "longitude latitude" ;
        float windNorthward(Location, WindEvent) ;
                windNorthward:_FillValue = 1.e+11f ;
                string windNorthward:long_name = "U, V-Component of Wind Quality Marker" ;
                string windNorthward:coordinates = "longitude latitude" ;
  } // group QualityMarker


```

**Comment by rmclaren** [comment]
Should be fixed now. Please try it. (the fix is in NCEPLIB-bufr feature/query_enhancements).

**Comment by YoulongXia-NOAA** [comment]
@emilyhcliu @rmclaren , as Emily has tested and our issues have fixed. I closed my issue #979. Thank you.

**Comment by rmclaren** [comment]
Please close this issue if it has been fixed.
